lib/json/ld/compaction.ex

```diff
@@ -36,7 +40,8 @@ defmodule JSON.LD.Compaction do
          compact_arrays \\ true
        )
 
-  # 1) If element is a scalar, it is already in its most compact form, so simply return element.
+  # 1) If element is a scalar, it is already in its most compact form, so simply
```

Comment:
Thanks for sharing your perspective of the maintainer, will be good lesson for me!
I do appreciate your time and your efforts at working on this library, and the only reason behind some aesthetic changes is the wish to contribute back as well during the main changes of PR.

Reply:
I see and assumed good will, but doing things along-the-way is generally not a good idea in PRs and commits. Do changes in separate commits and when they are unrelated in general even in separate PRs. What's also a good idea in general and would have saved us both a lot of time, would be to propose your changes upfront in an issue.

Reply:
Good idea, it should become my everyday practise.
Regarding to the next release I see still open issue related to expected order in test suite. I could try to fix that and that fix could be a candidate for the next release together with current changes.

Reply:
Ok, good that you ask now before-hand, so I have a chance to stop you from that. 

This is a non-trivial task. You really can't fix the ordering issue without fundamentally changing RDF.ex, since it is based on maps which are unordered data structures. But that's actually totally fine, since RDF graphs are unordered data structures. So, in order to really solve this a solution for the [graph isomorphism problem](https://en.wikipedia.org/wiki/Graph_isomorphism_problem) must be available in RDF.ex, so that we can do  isomorphism comparisons instead of equivalence comparisons during the execution of the test suite. I already started with an implementation of a solution for this a while ago by implementing this [spec for RDF dataset normalization](https://json-ld.github.io/normalization/spec/). But there are some problems in my implementation and I didn't have a chance to finish this, yet. 

If you really want, you could try to finish this implementation, which would help a lot. 
But this would still be a hard task. I could also provide you with some other issues you could tackle.

Reply:
Well, in that case I prefer to leave it as is because it's not critical for my current needs like it was with remote contexts.
To ask before-hand is useful, I see )